linkerd/app/outbound/src/lib.rs

```diff
@@ -68,9 +68,47 @@ impl Config {
     > + Unpin
            + Clone
            + Send {
+        let connect = svc::connect(self.proxy.connect.keepalive).into_inner();
         // Establishes connections to remote peers (for both TCP
         // forwarding and HTTP proxying).
-        svc::connect(self.proxy.connect.keepalive)
+        self.build_tcp_connect_with(connect, local_identity, metrics)
+    }
+
+    pub fn build_tcp_connect_with<C>(
```

Comment:
Is this really necessaary? Can we provide a service and check its target?

In other words, why do we actually care about validating the socket/connection behavior? It comes at the cost of... a lot of verbosity.